### PR TITLE
Document the `--report-stats` argument

### PR DESCRIPTION
**Pull request by jejo86**
Explained the meaning of the --report-stats argument.

Signed-off-by: jejo86 <28619134+jejo86@users.noreply.github.com>

### CODE REVIEW

File: docs/setup/installation.md [hunk 1]
```diff
@@ -233,7 +233,7 @@ python -m synapse.app.homeserver \
     --report-stats=[yes|no]
 ```
 
-... substituting an appropriate value for `--server-name`.
+... substituting an appropriate value for `--server-name` and choosing whether or not to report usage statistics (hostname, synapse version & uptime, total_users etc.) to the developers via the `--report-stats` argument.
```
Comment (squahtx) [inline docs/setup/installation.md:236]: Can we wrap long lines?

```suggestion
... substituting an appropriate value for `--server-name` and choosing whether
or not to report usage statistics (hostname, Synapse version, uptime, total
users, etc.) to the developers via the `--report-stats` argument.
```

**Review by squahtx — COMMENTED** [review]
Thanks for your contribution to the docs!

Could you [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off) on your change and add a [changelog entry](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog)?

**Comment by jejo86** [comment]
Changelog entry is added and pull request is signed off.

File: changelog.d/13029.doc [hunk 2]
```diff
@@ -0,0 +1 @@
+Argument report-stats explained
```
Comment (squahtx) [inline changelog.d/13029.doc:1]: CI is a bit picky about the changelog message. How about:
```suggestion
Add an explanation of the `--report-stats` argument to the docs.
```

**Review by squahtx — COMMENTED** [review]
Thanks for making the changes!

There's just one last tweak needed to make CI happy and then we're good to merge.

**Review by squahtx — APPROVED** [review]
LGTM! Thanks for your contribution!

**Pull request merged**